include/envoy/network/address.h

```diff
@@ -91,6 +91,11 @@ class Ip {
    * @return the version of IP address.
    */
   virtual IpVersion version() const PURE;
+
+  /**
+   * @return true if address is Ipv6 and Ipv4 compatibility is disabled, false otherwise
+   */
+  virtual bool v6only() const PURE;
```

Comment:
Can't this just be on the IPv6 interface? Why do we need it on the top level IP interface?

Reply:
At a high level, I didn't want to change the semantics of one of the existing socket functions which looks at "v6only()" property of the address to setsockopt(,IPV6_V6ONLY,). Not sure if this still makes sense after we finish moving everything to socket, but that's probably for a later time discussion.

So, with respect to the comment above, I did things this way because I didn't know how else to extract the field from the private IpHelper struct. Note though that I am somewhat C++ challenged :-(

I guess that one alternative would be to move the field to the Ipv6Helper struct and that should be then retrievable via a new function, e.g., address.ip().ipv6().v6only(). I didn't do this because I couldn't tell if somebody else relies on Ipv6Helper being nothing more than sockaddr_in6.

Reply:
> I guess that one alternative would be to move the field to the Ipv6Helper struct and that should be then retrievable via a new function, e.g., address.ip().ipv6().v6only(). I didn't do this because I couldn't tell if somebody else relies on Ipv6Helper being nothing more than sockaddr_in6.

Right this is what I would do. The helper is just an implementation of an interface and I think this can live on the IPv6 interface?

Reply:
Done!